BrainPortal/app/models/tool_config.rb

```diff
@@ -337,8 +338,17 @@ def container_rules #:nodoc:
     if self.container_engine.present? && ( self.containerhub_image_name.blank? && self.container_image_userfile_id.blank? )
       errors[:container_engine] = "a container hub image name or a container image userfile ID should be set when the container engine is set"
     end
+
+    if self.container_engine.present? && self.container_engine == "Singularity" 
+      if self.container_index_location.present? && ! /^[a-z0-9]+\:\/\/$/.match(self.container_index_location)
```

Comment:
The regex should be specified with the `/i` modifier. Also, it is better (IMHO) to use the syntax

```ruby
   $var !~ /regex/
```

instead of

```ruby
! /regex/.match(var)
```

Reply:
There are two places where this comment applies (the other one is about 10 lines below).